website/docs/plugins/ax_sweeper.md

```diff
@@ -70,15 +71,15 @@ banana_y:
+The `x` parameter takes on a "range" of integer values, in the interval `[-5, 5]`. The `y` parameter takes on a range of floating-point values in the interval `[-5, 10.1]`. In general, the plugin supports all the [Parameters](https://ax.dev/api/core.html?highlight=range#module-ax.core.parameter) that Ax supports. According to the [Ax documentation](https://ax.dev/api/service.html#ax.service.ax_client.AxClient.create_experiment), the required elements in the config are:
 
 * `name` - Name of the parameter. It is of type string.
 * `type` - Type of the parameter. It can take the following values: `range`, `fixed`, or `choice`.
 * `bounds` - Required only for the `range` parameters. It should be a list of two values, with the lower bound first.
 * `values` - Required only for the `choice` parameters. It should be a list of values.
 * `value` - Required only for the `fixed` parameters. It should be a single value. 
 
-Note that when using the config file, the parameter type (int, float, string, etc.) is set via the [`parameter_type` attribute](https://ax.dev/api/core.html?highlight=range#module-ax.core.parameter). One important thing to note is how mixed types (float + int) are handled. For `y` parameter, the range bounds were set to be `-5` to `10.1`. Both the values were upcasted to a float (irrespective of whether the range was set via the command line or the config file). In case the user provides the `parameter_type` attribute in the config, the attribute is not changed when type casting is done. If the user wants to sample integers in range `-5` to `5`, they need to specify the range as `-5:5` or `[-5, 5]` (in config). If they want to sample floats in range `-5` to `5`, they need to specify the range as `-5.0:5.0` or `[-5.0, 5.0]` (in config). The type casted values, and unchanged attributes are passed to the Ax Client. 
+Note that when using the config file, the parameter type (int, float, string, etc.) is set via the [`parameter_type` attribute](https://ax.dev/api/core.html?highlight=range#module-ax.core.parameter). One important thing to note is how mixed types (float + int) are handled. For the `y` parameter, the bounds were set to be `-5` to `10.1`. Both the values were upcasted to a float (irrespective of whether the range was set via the command line or the config file). In case the user provides the `parameter_type` attribute in the config, the attribute is not changed when type casting is done. If the user wants to sample integers in the interval `-5` to `5`, they need to specify the range as `interval(-5, 5)` (in the command line) or `[-5, 5]` (in config). If they want to sample floats in range `-5` to `5`, they need to specify the range as `interval(-5.0, 5.0)` (in the command line) or `[-5.0, 5.0]` (in config). The type casted values, and unchanged attributes are passed to the Ax Client.
```

Comment:
Intervals are always floating points, can Ax handler integer intervals (representing non contentious set?)

Reply:
Yeah. For example, the output for 

> Nice.
> 
> Side note:
> I abandoned the idea of compact keys in OmegaConf. This is a good chance to switch back to the more intuitive usage of dot and not _ to separate the path element in keys:
> 
> ```yaml
>       params:
>         banana.x:
>           type: range
>           bounds: [-5, 5]
> ```

Would it make sense as a separate issue ? Or should I fix in this?

Reply:
It does. For example, for the command: `python example/banana.py -m banana.x=interval(-5, 5) banana.y=interval(-5, 10.1)`

The output of a run looks like:

```
[HYDRA] AxSweeper is optimizing the following parameters:
banana.x: range=[-5, 5], type = int
banana.y: range=[-5.0, 10.1], type = float
ax.modelbridge.dispatch_utils: Using Bayesian Optimization generation strategy: GenerationStrategy(name='Sobol+GPEI', steps=[Sobol for 5 arms, GPEI for subsequent arms], generated 0 arm(s) so far). Iterations after 5 will take longer to generate due to model-fitting.
AxSweeper is launching 5 jobs
[HYDRA] Launching 5 jobs locally
[HYDRA]     #0 : banana.x=4 banana.y=-1.484
[__main__][INFO] - Banana_Function(x=4, y=-1.484)=30581.473
[HYDRA]     #1 : banana.x=3 banana.y=-3.653
[__main__][INFO] - Banana_Function(x=3, y=-3.653)=16014.261
[HYDRA]     #2 : banana.x=0 banana.y=9.409
[__main__][INFO] - Banana_Function(x=0, y=9.409)=8855.340
[HYDRA]     #3 : banana.x=-4 banana.y=2.059
[__main__][INFO] - Banana_Function(x=-4, y=2.059)=19459.063
[HYDRA]     #4 : banana.x=-3 banana.y=-1.338
[__main__][INFO] - Banana_Function(x=-3, y=-1.338)=10704.497
[HYDRA] New best value: 8855.340, best parameters: {'banana.x': 0, 'banana.y': 9.409}
```

Note the values of x are all integers.

Reply:
Does Nevergrad has similar capability?

Currently intervals are always floating points, but we can change that.
an alternative is to utilize tag here:
```python
tag(int,interval(1,2))
```

Reply:
What is the difference between interval(1,10) and range (1,10) ?

Reply:
one thing that could work well is to default interval to floating point (even if input is ints), but to allow casting it to int:
```
interval(1,10) -> float interval, 1 <= x < 10
int(interval(1,10)) -> int interval, 1 <= x < 10
```

Reply:
My interpretation for the Ax-plugin is as follows:

1. Hydra interval maps to Ax range.
2. Hydra interval is always continous while Ax range can be discrete or continous.
3. Hydra range maps to Ax choice. 
4. Ax-plugin would map hydra interval (1, 10) to Ax range (1, 10), with type as int. 
5. Ax-plugin would map hydra range (1, 10) to Ax choice [1, 2, ...9], with type as int.

Reply:
Not sure how Nevergrad handles it but `tag(int,interval(1,2))` seems complex. The current behaviour is faithful to the previous behaviour.

Reply:
interval(1,10) is IntervalSweep(1.0, 10.0).
There is currently no such thing as an integer interval in Hydra.

Reply:
To keep Hydra 1.0 on track, I am not going to change that.
Use tag to support this for now. we can revisit once we get some feedback and have better understanding of the usefulness of this.
I believe the correct interpretation of an interval is a floating point interval even if the numbers specified are integers.

We can either add support for casting an interval (which is not supported now because it's always a float), or support it via the tag mechanism.

Reply:
I agree with `My concern is that in 95% of the cases the user actually wants a regular floating point interval, so that case should be easy and intuitive.`. What is the best way to go ahead with that ?

Reply:
what I said.
```
interval(1,10) -> float interval, 1 <= x < 10
int(interval(1,10)) -> int interval, 1 <= x < 10

or
tag(int,interval(1,10))
```

I would rather that we added int casting support to interval.

Reply:
Looks like Nevergrad does not have a similar int interval (https://facebookresearch.github.io/nevergrad/parametrization_ref.html#nevergrad.p.Scalar).

Reply:
file a followup task to add support to casting interval to int.

Reply:
Okay will get to it towards the end of the day

Reply:
I still think that using tags will be somewhat complicated for the end user and will hurt adoption of the plugin. Would the following be a reasonable compromise:

All intervals will be float intervals and if the user wants to use intervals with integer values, they can use range (where we will typecast the values to be int if start, step and stop, all are int

Reply:
Ahh I think one issue with approach is that we are relying on an internal detail of Ax - that Int Range and choice are the same thing in Ax. I will go with your original suggestion of using tags.

Reply:
One more place I could use some clarification is the following:

When we specify the range via the config file (say [-1, 1]) and param type is not explicitly specified, we interpret the type to be int. Do we want to change these semantics as well?

Reply:
For now do something reasonable with the config, I think we will need to have a pass at standardizing the how the parameters are configured and how the config and the command line interacts.

> I still think that using tags will be somewhat complicated for the end user and will hurt adoption of the plugin

I strongly disagree.
In this context, tags are used to mark an interval as a very special and unusual interval of int types. In my opinion it's also pretty useless but that's remains to be seen.
If the user actually needs a discrete range they should use range() and not interval().

Also, advanced users will learn how to use tags.